predicators/approaches/bridge_policy_approach.py

```diff
@@ -365,10 +366,79 @@ def __init__(self,
                          action_space, train_tasks, task_planning_heuristic,
                          max_skeletons_optimized)
         self._maple_initialized = False
+        if task_planning_heuristic == "default":
+            task_planning_heuristic = CFG.sesame_task_planning_heuristic
+        self._task_planning_heuristic = task_planning_heuristic
         self._trajs: List[LowLevelTrajectory] = []
+        self.CanPlan = Predicate("CanPlan", [], self._Can_plan)
+        self.CallPlanner = utils.SingletonParameterizedOption(
+            "CallPlanner",
+            types=None,
+            policy=self.call_planner_policy,
+            params_space=Box(0, len(train_tasks), (1, )),
+        )
+        initial_options.add(self.CallPlanner)
+        self._initial_options = initial_options
         self.mapleq=MapleQApproach(self._get_current_predicates(), \
                                    self._initial_options, self._types, \
                                     self._action_space, self._train_tasks)
+        self._current_control = ""
+        option_policy = self._get_option_policy_by_planning(
+            self._train_tasks[0], CFG.timeout)
+        self._current_policy = utils.option_policy_to_policy(
+            option_policy,
+            max_option_steps=CFG.max_num_steps_option_rollout,
+            raise_error_on_repeated_state=True,
+        )
+        self._bridge_called_state = State(data={})
+        self._policy_logs: List[str] = []
```

Comment:
as far as i can tell, you're only ever setting this value, never reading from it. What's the point of this? Are you logging/using it somewhere?

Reply:
self._bridge_called_state is used for the precondition of CallPlanner, _Can_plan, which is False if self._bridge_called_state is the same as our current state, and True if they are different.

Reply:
sorry - i meant `self._policy_logs`

Reply:
`self._policy_logs` is used in `learn_from_interaction_results` to determine which actions were executed by bridge and which actions were executed by planner, so that bridge policy only learns from the actions that itself takes